src/Fluid/Flu_BoundaryCondition_User.cpp

```diff
@@ -200,9 +203,20 @@ void Flu_BoundaryCondition_User( real *Array, const int NVar_Flu, const int Arra
       v2 = NVar_Flu;
 
 #     if   ( MODEL == HYDRO )
+#     ifdef SRHD
+      real LorentzFactor, Prim[NCOMP_FLUID];
+
+      Hydro_Con2Pri( BVal, Prim, (real)NULL_REAL, NULL_BOOL, NULL_INT, NULL, NULL_BOOL,
+                     (real)NULL_REAL, EoS_DensEint2Pres_CPUPtr, EoS_DensPres2Eint_CPUPtr,
+                     EoS_GuessHTilde_CPUPtr, EoS_HTilde2Temp_CPUPtr, EoS_AuxArray_Flt,
+                     EoS_AuxArray_Int, h_EoS_Table, NULL, &LorentzFactor );
+
+      if ( PrepLrtz )   Array3D[ v2 ++ ][k][j][i] = LorentzFactor;
+#     else
       if ( PrepVx   )   Array3D[ v2 ++ ][k][j][i] = BVal[MOMX] / BVal[DENS];
```

Comment:
@zengbs just a quick question--are lines 216-218 which prepare the velocity still valid in SR?

Reply:
Or I guess they are not even called if `SRHD` is defined. So we don't need to prepare the velocities here?

Reply:
@jzuhone Actually, these lines of preparing velocities are currently not used in either HD or SRHD. Only momenta are used.